kytos/core/auth.py

```diff
@@ -44,6 +50,83 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all topology related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)]),
+        ]
+        for collection, keys in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data):
+        """Create user to database"""
+        utc_now = datetime.datetime.utcnow()
+        try:
+            data = {
+                "username": user_data["username"],
+                "email": user_data["email"],
+                "password": user_data["password"],
+                "inserted_at": utc_now,
+                "updated_at": utc_now,
+            }
+            user = UserDoc(**data)
+            result = self.db.users.insert_one(user.dict())
+        except (ValueError, KeyError, Exception) as e:
```

Comment:
This `try` `except` here is just re-raising, so it's no handling much or re-transforming exceptions. Since we're not going to introduce new exceptions to make it easier to maintain and evolve, I'll recommend to you to handle `pydantic.ValidationError` and MongoDB `DuplicateKeyError` (once you add the unique index) at the call site where this method is being called.

Reply:
Changed.